producer/producer_sf.go

```diff
@@ -266,7 +266,7 @@ func SearchSFlowSamplesConfig(samples []interface{}, config *SFlowProducerConfig
 				flowMessage.DstNet = recordData.DstMaskLen
 			case sflow.ExtendedGateway:
 				ipNh = recordData.NextHop
-				flowMessage.NextHop = ipNh
+				flowMessage.BgpNextHop = ipNh
```

Comment:
Why adding a distinct field? Are there cases where NextHop (from ExtendedRouter) and BgpNextHop (from ExtendedGateway) are different?

Reply:
Yup, that was interesting find actually.

https://www.cisco.com/en/US/technologies/tk648/tk362/technologies_white_paper09186a00800a3db9.html

![image](https://user-images.githubusercontent.com/327717/121135503-dd349f00-c834-11eb-8c7b-c3ee6db1e74e.png)

Should be covered by the test where `bgp next hop ip` is `5.5.5.5` and next hop ip is `9.9.9.9`